src/askui_runner/modules/core/infrastructure/files/askui.py

```diff
@@ -104,136 +86,121 @@ def sync(
-
         # List remote files
         remote_files: Dict[str, FileDto] = {}
-        list_objects_response = self._list_object_recursive(remote_dir_path)
-        for content in list_objects_response.data:
-            if any(
-                re.match(pattern, content.path)
-                for pattern in self.HIDDEN_FILES_PATTERNS
-            ):
-                continue
-            remote_files[content.path[len(remote_dir_path) :].lstrip("/")] = content
+
+        for file in self._list_remote_objects(remote_dir_path):
+            remote_files[file.path[len(remote_dir_path) :].lstrip("/")] = file
 
         # List local files
         local_files = self._list_local_files(local_dir_path)
 
         # Create lookup table
         all_paths = set(local_files.keys()) | set(remote_files.keys())
 
-        # Action mapper
-        action_mapper: Dict[str, Callable[..., None]] = {
-            "upload": lambda local_file_path, remote_file_path: (
-                logging.info(f"Dry: Upload {local_file_path} to cloud")
-                if dry
-                else self._upload_file(local_file_path, remote_file_path)
-            ),
-            "download": lambda remote_file_url, remote_file_last_modified, local_path: (
-                logging.info(f"Dry: Download file to {local_path}")
-                if dry
-                else self._download_file(
-                    remote_file_url, local_path, remote_file_last_modified
-                )
-            ),
-            "delete_local": lambda local_file_path: (
-                logging.info(f"Dry: Delete local file {local_file_path}")
-                if dry
-                else os.remove(local_file_path)
-            ),
-            "delete_remote": lambda remote_file_path: (
-                logging.info(f"Dry: Delete remote file {remote_file_path}")
-                if dry
-                else self._delete_file(remote_file_path)
-            ),
-            "skip": lambda relative_path: logging.info(
-                f"Skip {relative_path} (no changes)"
-            ),
-        }
-
         for relative_path in sorted(all_paths):
-            local_file: Union[LocalFileDto | None] = local_files.get(relative_path)
+            local_file: Union[FileDto | None] = local_files.get(relative_path)
             remote_file: Union[FileDto | None] = remote_files.get(relative_path)
             remote_file_path = f"{remote_dir_path}/{relative_path}".replace("\\", "/")
 
             # File exists in both locations
             if local_file and remote_file:
-                local_mtime = local_file.last_modified
-                remote_mtime = (
-                    datetime.strptime(remote_file.last_modified, "%Y-%m-%dT%H:%M:%SZ")
-                    .replace(tzinfo=timezone.utc)
-                    .timestamp()
-                )
+                local_mtime = local_file.last_modified.timestamp()
+                remote_mtime = remote_file.last_modified.timestamp()
```

Comment:
I think you can directly compare `datetime` (including `AwareDatetime`) using `>` so there is no need to get the timestamps from my perspective.